scripts/TestHarness/testers/RAVENImageDiff.py

```diff
@@ -56,60 +61,108 @@ def diff(self):
-      pct_num_diff = only_diffs.size/float(diff.size)
-      if pct_num_diff > self.__options['rel_err']:
-        self.__messages += 'Difference between images is too large:'+\
-          ' %2.2f pct (allowable: %2.2f)' %(100*pct_num_diff,
-                                            100*self.__options['rel_err'])
-        self.__same = False
-    return (self.__same, self.__messages)
+      #read in files
+      if files_read:
+        if not correctImport:
+          self.__message += 'ImageDiff cannot run with scipy version less '+\
+            'than 0.15.0, and requires the PIL installed; scipy version is '+\
+            str(scipy.__version__)
+          self.__same = False
+          return(self.__same, self.__message)
+        try:
+          # RAK - The original line...
+          # test_image = imread(open(test_filename,'r'))
+          # ...didn't work on Windows Python because it couldn't sense the file type
+          test_image = imread(test_filename)
+        except IOError:
+          self.__message += 'Unrecognized file type for test image in scipy.imread: '+test_filename
+          files_read = False
+          return (False, self.__message)
+        try:
+          # RAK - The original line...
+          # gold_image = imread(open(gold_filename,'r'))
+          # ...didn't work on Windows Python because it couldn't sense the file type
+          gold_image = imread(gold_filename)
+        except IOError:
+          files_read = False
+          self.__message += 'Unrecognized file type for test image in scipy.imread: '+gold_filename
+          return (False, self.__message)
+        #first check dimensionality
+        if gold_image.shape != test_image.shape:
+          self.__message += 'Gold and test image are not the same shape: '+\
+            str(gold_image.shape)+', '+str(test_image.shape)
+          self.__same = False
+          return (self.__same, self.__message)
+        #pixelwise comparison
+        #TODO in the future we can add greyscale, normalized coloring, etc.
+        # For now just do raw comparison of right/wrong pixels
+        diff = gold_image - test_image
+        only_diffs = diff[abs(diff) > self.__zero_threshold]
+        pct_num_diff = only_diffs.size/float(diff.size)
+        if pct_num_diff > self.__rel_err:
+          self.__message += 'Difference between images is too large:'+\
+            ' %2.2f pct (allowable: %2.2f)' %(100*pct_num_diff,\
+                                            100*self.__rel_err)
+          self.__same = False
+    return (self.__same, self.__message)
+
+
+class ImageDiffer(Differ):
+  """
+  This is the class to use for handling the parameters block.
+  """
+
+  @staticmethod
+  def get_valid_params():
+    """
+      Returns the valid parameters for this class.
+      @ In, None
+      @ Out, params, _ValidParameters, return the parameters.
+    """
+    params = Differ.get_valid_params()
+    params.add_param('rel_err', '', 'Relative Error for image files')
```

Comment:
It might be worth considering naming this something other than rel_err since it looks like this is the fraction of pixels that can be different.